components/cli/pkg/kubernetes/instance.go

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2019 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package kubernetes
+
+import (
+	"fmt"
+
+	errorpkg "github.com/cellery-io/sdk/components/cli/pkg/error"
+)
+
+func GetInstanceAvailability(instanceName string) (bool, error) {
```

Comment:
Shall we rename this to "InstanceAvailable" or "IsInstanceAvailable"

Reply:
Also IMO this should be an interface function in KubeCli interface. CelleryKubeCli and MockKubeCli should implement this function.

Reply:
> Shall we rename this to "InstanceAvailable" or "IsInstanceAvailable"
GetInstanceAvailability is used because the function is returning the instance availability, IMHO GetInstanceAvailability  is better than InstanceAvailable or IsInstanceAvailable in here

> Also IMO this should be an interface function in KubeCli interface. CelleryKubeCli and MockKubeCli should implement this function.

Since this function is not directly calling to kubectl (uses GetCell, GetInstance methods) do we have a use-case to make this function as an interface? MockKubeCli is not intended for dealing with this function.

Reply:
This function is indirectly calling kubectl via "GetCell" and "GetComposite" functions. The use case is to check the availability of a mocked instance (bellow comment).

Reply:
Got it. Thank you 👍